### PR TITLE
add a functional parameter OptionMatchOR

### PR DESCRIPTION
**Pull request by darkonie**
OptionMatchOR takes a list of EntryMatch objects and and uses OR between
them.

### CODE REVIEW

File: dcos-log/journal/reader/config.go [hunk 1]
```diff
@@ -52,6 +52,23 @@ func OptionMatch(m []JournalEntryMatch) Option {
...
+// arguments []JournalEntryMatch.
+func OptionMatchOR(m []JournalEntryMatch) Option {
+	return func(r *Reader) error {
+		if r.Journal == nil {
+			return ErrUninitializedReader
+		}
+
+		for _, match := range m {
+			r.Journal.AddMatch(match.String())
+			r.Journal.AddDisjunction()
```
Comment (collinvandyck) [inline dcos-log/journal/reader/config.go]: Is this the kind of thing where `AddDisjunction` only needs to be applied to slice indices [0..(n-1)) ?  In other words, will this possibly leave a dangling OR at the end of the match?

Comment (darkonie) [inline dcos-log/journal/reader/config.go]: yeah, that's intensional, if i never need to ad logical and to a,b,c it'd be like `a|b|c| a1&b1`
if we don't have the OR at the end, we implicitly say AND (which is wrong)
https://www.freedesktop.org/software/systemd/man/sd_journal_add_match.html

Comment (collinvandyck) [inline dcos-log/journal/reader/config.go]: Just curious, are we able to have parenthetical boolean expressions?

Comment (darkonie) [inline dcos-log/journal/reader/config.go]: nope :)

**Pull request merged**